paddle/cinn/hlir/framework/group_scheduler.cc

```diff
@@ -0,0 +1,1224 @@
+        load_indice_value, CudaBindInfo::kCudaThreadAndSerial, load_block_name);
+    auto store_block_coefficient_and_range = GetCoefficientAndRange(
+        store_indice_value, ir::ForType::GPUBlock, store_block_name);
+    auto load_block_coefficient_and_range = GetCoefficientAndRange(
+        load_indice_value, ir::ForType::GPUBlock, load_block_name);
+    VLOG(6) << "store_indice_value: " << store_indice_value;
+    VLOG(6) << "load_indice_value: " << load_indice_value;
+    VLOG(6) << "store_block_name: " << store_block_name;
+    VLOG(6) << "load_block_name: " << load_block_name;
+    VLOG(6) << "store_block_overall_range = (" << store_block_overall_range.min
+            << ", " << store_block_overall_range.max << ")";
+    VLOG(6) << "load_block_overall_range = (" << load_block_overall_range.min
+            << ", " << load_block_overall_range.max << ")";
+    VLOG(6) << "store_thread_and_serial_overall_range = ("
+            << store_thread_and_serial_overall_range.min << ", "
+            << store_thread_and_serial_overall_range.max << ")";
+    VLOG(6) << "load_thread_and_serial_overall_range = ("
+            << load_thread_and_serial_overall_range.min << ", "
+            << load_thread_and_serial_overall_range.max << ")";
+    VLOG(6) << "store_block_coefficient_and_range[0] = <"
+            << store_block_coefficient_and_range[0].first << ", ("
+            << store_block_coefficient_and_range[0].second.min << ", "
+            << store_block_coefficient_and_range[0].second.max << ")>";
+    VLOG(6) << "load_block_coefficient_and_range[0] = <"
+            << load_block_coefficient_and_range[0].first << ", ("
+            << load_block_coefficient_and_range[0].second.min << ", "
+            << load_block_coefficient_and_range[0].second.max << ")>";
+    VLOG(6) << "store_block_coefficient_and_range[1] = <"
+            << store_block_coefficient_and_range[1].first << ", ("
+            << store_block_coefficient_and_range[1].second.min << ", "
+            << store_block_coefficient_and_range[1].second.max << ")>";
+    VLOG(6) << "load_block_coefficient_and_range[1] = <"
+            << load_block_coefficient_and_range[1].first << ", ("
+            << load_block_coefficient_and_range[1].second.min << ", "
+            << load_block_coefficient_and_range[1].second.max << ")>";
+    VLOG(6) << "store_block_coefficient_and_range[2] = <"
+            << store_block_coefficient_and_range[2].first << ", ("
+            << store_block_coefficient_and_range[2].second.min << ", "
+            << store_block_coefficient_and_range[2].second.max << ")>";
+    VLOG(6) << "load_block_coefficient_and_range[2] = <"
+            << load_block_coefficient_and_range[2].first << ", ("
+            << load_block_coefficient_and_range[2].second.min << ", "
+            << load_block_coefficient_and_range[2].second.max << ")>";
+    return !(store_block_overall_range.min <= load_block_overall_range.min &&
```

Comment:
Same as above

Reply:
After implementing auxiliary data structures such as IntegerSet and MultiDimIntegerSet, we can re implement this function to simplify these ugly codes. I have added a TODO above the implementation of function AllocateStorage().